stream_in.c

```diff
@@ -160,9 +161,52 @@ int in_set_parameters(struct audio_stream *stream, const char *kv_pairs)
 
 char * in_get_parameters(const struct audio_stream *stream, const char *keys)
 {
+    x_stream_in_t *xin = (x_stream_in_t*)stream;
+    struct str_parms * request_keys;
+    struct str_parms * response_pairs;
+    char temp_str[256];
+    bool have_response = false;
+    char *result_str = NULL;
+
+    pthread_mutex_lock(&xin->lock);
     LOG_FN_NAME_WITH_ARGS("(%p, '%s')", stream, keys);
-    /* TODO To implement */
-    return strdup("");
+
+    request_keys = str_parms_create_str(keys);
+    if (request_keys == NULL) {
+        pthread_mutex_unlock(&xin->lock);
+        return strdup("");
+    }
+
+    response_pairs = str_parms_create();
+    if (response_pairs == NULL) {
+        pthread_mutex_unlock(&xin->lock);
+        return strdup("");
+    }
+
+    if (str_parms_has_key(request_keys, AUDIO_PARAMETER_STREAM_SUP_FORMATS)) {
+        strcpy(temp_str, "AUDIO_FORMAT_PCM_16_BIT");
+        str_parms_add_str(response_pairs, AUDIO_PARAMETER_STREAM_SUP_FORMATS, temp_str);
```

Comment:
Maybe it is a good point to analyze return values from all str_params_* calls.

Reply:
Done.